### ISSUE TITLE
Padding of 255 for label during training

### ISSUE THREAD

**Issue by egginsect** [nicolov/segmentation_keras#17]
Hi,
For Pascal VOC, there are 21 classes including background. To my knowledge the padding of 255 is not belong to those 21 class. How does keras handling the padded pixel during training. Will those pixels be just ignored?

**Comment by egginsect** [comment]
I've found that it is the don't care label, which means the loss of the pixel labeled as 255 is zero.

**Comment by nicolov** [comment]
Caffe has an _ignore_label_ flag for cases like this, but keras doesn't (at least when I last looked at this). For this code, I noticed that only a few images actually had any 255 pixels, so I just skip them like [this](https://github.com/nicolov/segmentation_keras/blob/master/utils/image_reader.py#L202)
